treeherder/model/derived/jobs.py

```diff
@@ -1453,9 +1453,16 @@ def _load_ref_and_job_data_structs(
             tls_list = error_summary.get_artifacts_that_need_bug_suggestions(
                 artifacts)
             async_artifact_list.extend(tls_list)
-            ArtifactsModel.populate_placeholders(artifacts,
-                                                 artifact_placeholders,
-                                                 job_guid)
+
+            # need to add job guid to artifacts, since they likely weren't
+            # present in the beginning
+            for artifact in artifacts:
+                if not all(k in artifact for k in ("name", "type", "blob")):
+                    raise JobDataError(
+                        "Artifact missing properties: {}".format(artifact))
```

Comment:
This is kind of a new thing, before we would silently drop these artifacts if these weren't specified when converting them to the old placeholder list. I think it's better to raise an error.